docs/pandas.rst

```diff
@@ -0,0 +1,333 @@
+   >>> print(df_init)
+      speed  mech power  torque  rail pressure  fuel flow rate  fluid power
+   0   1000         NaN      10           1000              10          NaN
+   1   1100         NaN      10  1000000000000              10          NaN
+   2   1200         NaN      10           1000              10          NaN
+   3   1200         NaN      10           1000              10          NaN
+
+Then we add a column header which contains units information
+
+.. doctest::
+
+   >>> units = ["rpm", "kW", "N m", "bar", "l/min", "kW"]
+   >>> df_to_save = df_init.copy()
+   >>> df_to_save.columns = pd.MultiIndex.from_arrays([df_init.columns, units])
+   >>> print(df_to_save)
+     speed mech power torque  rail pressure fuel flow rate fluid power
+       rpm         kW    N m            bar          l/min          kW
+   0  1000        NaN     10           1000             10         NaN
+   1  1100        NaN     10  1000000000000             10         NaN
+   2  1200        NaN     10           1000             10         NaN
+   3  1200        NaN     10           1000             10         NaN
+
+Now we save this to disk as a csv to give us our starting point.
+
+.. doctest::
+
+   >>> test_csv_name = "pandas_test.csv"
+   >>> df_to_save.to_csv(test_csv_name, index=False)
+
+Now we are in a position to read the csv we just saved. Let's start by reading the file with units as a level in a multiindex column.
+
+.. doctest::
+
+   >>> df = pd.read_csv(test_csv_name, header=[0,1])
+   >>> print(df)
+     speed mech power torque  rail pressure fuel flow rate fluid power
+       rpm         kW    N m            bar          l/min          kW
+   0  1000        NaN     10           1000             10         NaN
+   1  1100        NaN     10  1000000000000             10         NaN
+   2  1200        NaN     10           1000             10         NaN
+   3  1200        NaN     10           1000             10         NaN
+
+Then use the DataFrame's `pint.quantify` method to convert the columns from `np.ndarray`s to PintArrays, with units from the bottom column level.
+
+.. doctest::
+
+   >>> df_ = df.pint.quantify(ureg, level=-1)
+   >>> print(df_)
+   0  1000.0 revolutions_per_minute  nan kilowatt  10.0 meter * newton
+   1  1100.0 revolutions_per_minute  nan kilowatt  10.0 meter * newton
+   2  1200.0 revolutions_per_minute  nan kilowatt  10.0 meter * newton
+   3  1200.0 revolutions_per_minute  nan kilowatt  10.0 meter * newton
+
+            rail pressure       fuel flow rate   fluid power
+   0           1000.0 bar  10.0 liter / minute  nan kilowatt
+   1  1000000000000.0 bar  10.0 liter / minute  nan kilowatt
+   2           1000.0 bar  10.0 liter / minute  nan kilowatt
+   3           1000.0 bar  10.0 liter / minute  nan kilowatt
+
+
+
+.. doctest::
+
+   >>> from os.path import join
+   >>> df=pd.read_csv(join("..", "pint", "testsuite", "test-data", "pint_test_data.csv"), header=[0,1])
```

Comment:
I think here should be now just: pandas_test.csv
So that people load what they have just saved

Reply:
oops that was meant to be deleted, good pick up!